controllers/admin/AdminOrdersController.php

```diff
@@ -1334,6 +1334,8 @@ public function postProcess()
             if ($this->tabAccess['edit'] === '1') {
                 if (!Tools::getValue('discount_name')) {
                     $this->errors[] = Tools::displayError('You must specify a name in order to create a new discount.');
+                } elseif (!Tools::getValue('discount_value') || ((float)Tools::getValue('discount_value') <= 0)) {
```

Comment:
@eternoendless should I put (Tools::getValue('discount_value')) in a variable to prevent repeating the sentence twice?

Reply:
I don't think !Tools::getValue('discount_value') is needed at all, since (float)Tools::getValue('discount_value') <= 0) cover the "0"/empty value

Reply:
you've got a point ;) ok I will change it. Thanks @jocel1 :)